### PR TITLE
move node.foo style config to node.attr.foo

### PR DESCRIPTION
**Pull request by nslater**
https://trello.com/c/ph1jP3js/1225

### CODE REVIEW

File: blackbox/docs/configuration/cluster.txt [hunk 1]
```diff
@@ -634,12 +634,12 @@ across generic attributes associated with nodes.
 
   Define node attributes which will be used to do awareness based on the
   allocation of a shard and its replicas. For example, let's say we have
-  defined an attribute ``rack_id`` and we start 2 nodes with ``node.rack_id``
-  set to rack_one, and deploy a single table with 5 shards and 1 replica. The
-  table will be fully deployed on the current nodes (5 shards and 1 replica
-  each, total of 10 shards).
+  defined an attribute ``rack_id`` and we start 2 nodes with
+  ``node.attr.rack_id`` set to rack_one, and deploy a single table with 5
```
Comment (mxm) [inline blackbox/docs/configuration/cluster.txt]: Should we also add this setting to the configuration page?

Comment (nslater) [reply blackbox/docs/configuration/cluster.txt]: there's a lot of stuff we need to improve in the docs re *what* is documented. but I'd prefer to tackle them in a different pr

File: app/src/main/dist/config/crate.yml [hunk 2]
```diff
@@ -216,7 +216,7 @@
 
 # Node names are generated dynamically on startup, so you're relieved
 # from configuring them manually. You can tie this node to a specific name:
-#node.name: "Piz Buin"
+#node.attr.name: "Piz Buin"
```
Comment (mfussenegger) [inline app/src/main/dist/config/crate.yml]: This is still `node.name` without `attr`

File: app/src/main/dist/config/crate.yml [hunk 3]
```diff
@@ -250,13 +250,13 @@
...
-# an example:
-#node.rack: rack314
+# An attribute is a simple key value pair, similar to node.attr.key: value,
+# here is an example:
+#node.attr.rack: rack314
 
 # By default, multiple nodes are not allowed to start from the same
 # installation location. To allow it, set the following to a higher value:
-#node.max_local_storage_nodes: 1
+#node.attr.max_local_storage_nodes: 1
```
Comment (mfussenegger) [inline app/src/main/dist/config/crate.yml]: Still without `attr`

File: app/src/main/dist/config/crate.yml [hunk 4]
```diff
@@ -791,7 +791,7 @@
 
 # Defines whether SQL statements resulting in modification operations are
 # allowed on this node.
-#node.sql.read_only: false
+#node.attr.sql.read_only: false
```
Comment (mfussenegger) [inline app/src/main/dist/config/crate.yml]: Still without `attr`

File: app/src/main/dist/config/crate.yml [hunk 5]
```diff
@@ -806,11 +806,11 @@
 # a query.
 
 # Specifies the limit for the query breaker.
-#node.breaker.query.limit: 60%
+#node.attr.breaker.query.limit: 60%
```
Comment (mfussenegger) [inline app/src/main/dist/config/crate.yml]: I think this setting doesn't exist anymore. There is a `indices.breaker.query.limit` instead

File: app/src/main/dist/config/crate.yml [hunk 6]
```diff
@@ -806,11 +806,11 @@
...
 
 # A constant that all query data estimations are multiplied with to
 # determine a final estimation.
-#node.breaker.query.overhead: 1.09
+#node.attr.breaker.query.overhead: 1.09
```
Comment (mfussenegger) [inline app/src/main/dist/config/crate.yml]: See previous comment, I think this is now in the `indices` namespace.

File: blackbox/docs/configuration/node.txt [hunk 7]
```diff
@@ -95,12 +95,12 @@ Nodes that do not contain *data* and are not eligible as *master* are called
    yaml
 
    node:
-     client: true
+     attr.client: true
```
Comment (mfussenegger) [inline blackbox/docs/configuration/node.txt]: See comment above - `node.client` has been removed.

File: blackbox/docs/configuration/node.txt [hunk 8]
```diff
@@ -95,12 +95,12 @@ Nodes that do not contain *data* and are not eligible as *master* are called
...
 
 Read-only node
 --------------
 
-**node.sql.read_only**
+**node.attr.sql.read_only**
```
Comment (mfussenegger) [inline blackbox/docs/configuration/node.txt]: Still used without `attr`

File: blackbox/docs/release_notes/2.0.0.txt [hunk 9]
```diff
@@ -30,8 +30,12 @@ Breaking Changes
 
    If you disable this setting, CrateDB will run with the standard feature set.
 
- - The default value of the ``node.max_local_storage_nodes`` node setting has
-   been changed to ``1`` to prevent running multiple nodes on the same data
+ - All ``node.*`` style attributes must now be written as ``node.attr.*``, with
```
Comment (mfussenegger) [inline blackbox/docs/release_notes/2.0.0.txt]: Maybe rephrase this a bit.

All *custom* ``node.*`` style attributes must now be written as [...] to
distinguish them from core node attributes like ``node.name``.

The point is that anything in `node.attr` is *not* validated, but rather it's a
custom bag of attributes one can attach to a node. The keys/values mostly don't
matter.

(Except for how it can be used to control routing allocation a bit - see our
multi-zone setup documentation).

All `node.XY` settings are validated on startup and the node won't start up
if the setting is invalid or there is a typo in the name. E.g.
    `node.sql.read_only=foobar` will fail because it expects a boolean value.
    Similar `node.nme` will cause a failure because the setting doesn't exist.

File: sandbox/crate/config/crate.yml [hunk 10]
```diff
@@ -4,7 +4,7 @@ stats.enabled: true
 # bind to localhost, mainly to prevent the node from dying due to bootstrap checks (HEAP checks)
 network.host: _local_
 license.enterprise: true
-node.max_local_storage_nodes: 3
+node.attr.max_local_storage_nodes: 3
```
Comment (mfussenegger) [inline sandbox/crate/config/crate.yml]: without `attr`

File: app/src/main/dist/config/crate.yml (hunk 2 above)
Comment (mxm) [reply app/src/main/dist/config/crate.yml]: `node.name` is still the same. https://www.elastic.co/guide/en/elasticsearch/reference/current/important-settings.html#node.name

File: blackbox/docs/configuration/node.txt [hunk 11]
```diff
@@ -14,7 +14,7 @@ Node Specific Settings
 
   The name of the CrateDB cluster the node should join to.
 
-**node.name**
+**node.attr.name**
```
[the remark this answers is not in the document]
Comment (mxm) [reply blackbox/docs/configuration/node.txt]: https://www.elastic.co/guide/en/elasticsearch/reference/current/important-settings.html#node.name

**Comment by mxm** [comment]
Thanks for spotting the missing prefixes, @mfussenegger.

File: blackbox/docs/configuration/node.txt [hunk 12]
```diff
@@ -50,13 +50,13 @@ to differentiate nodes upon startup:
 
   Whether or not this node will store data.
 
-**node.client**
+**node.attr.client**
```
[the remark this answers is not in the document]
Comment (nslater) [reply blackbox/docs/configuration/node.txt]: do we need to note this in the release notes somewhere or can I just delete?

File: blackbox/docs/configuration/node.txt [hunk 13]
```diff
@@ -50,13 +50,13 @@ to differentiate nodes upon startup:
...
   | *Default:* ``false``
   | *Runtime:* ``no``
 
   Shorthand for: ``node.data=false`` and ``node.master=false``.
 
-**node.local**
+**node.attr.local**
```
[the remark this answers is not in the document]
Comment (nslater) [reply blackbox/docs/configuration/node.txt]: do we need to note this in the release notes somewhere or can I just delete?

File: blackbox/docs/configuration/node.txt (hunk 12 above)
[the remark this answers is not in the document]
Comment (mfussenegger) [reply blackbox/docs/configuration/node.txt]: It was part of the 2.0 changes - so yep I guess it would make sense to add a short note there.

File: app/src/main/dist/config/crate.yml (hunk 5 above)
Comment (nslater) [reply app/src/main/dist/config/crate.yml]: when did this change? I would like to put some notes in the release notes

Comment (mfussenegger) [reply app/src/main/dist/config/crate.yml]: This has been changed a long time ago - back to 0.50.0: https://github.com/crate/crate/commit/927824ed96d747036b269c3eebc77975a540c163

Comment (nslater) [reply app/src/main/dist/config/crate.yml]: okay that's far enough back that we don't even have release notes on master so I can't even add breaking changes

I could go back and edit the 0.50 branch if you think it's necessary?

File: blackbox/docs/configuration/node.txt [hunk 14]
```diff
@@ -58,52 +59,21 @@ to differentiate nodes upon startup:
...
-     data: true
-     master: false
-
-*Master-only*-nodes that do not contain *data* but are able to become *master*
-can be used to separate cluster-management loads from the query execution loads:
-
-.. code-block::
-   yaml
+Using different combinations of these two settings, you can create four
+different types of node. Each type of node is differentiate by what types of
```
Comment (mfussenegger) [inline blackbox/docs/configuration/node.txt]: is differentiate*d* ?

File: blackbox/docs/configuration/node.txt [hunk 15]
```diff
@@ -58,52 +59,21 @@ to differentiate nodes upon startup:
...
-
-   node:
-     client: true
++---------------+----------------------------+------------------------------+
+|               | **Master**                 | **No Master**                |
++---------------+----------------------------+------------------------------+
+| **Data**      | Handles all loads.         | Handles request handling and |
+|               |                            | query execution loads.       |
++---------------+----------------------------+------------------------------+
+| **No Data**   | Handles cluster management | Handles request handling     |
```
Comment (mfussenegger) [inline blackbox/docs/configuration/node.txt]: Maybe we should clarify here a bit. Nodes marked as `node.master`  *can* do cluster state management, but only one node at a time is elected as *the* master.

File: blackbox/docs/configuration/node.txt (hunk 14 above)
Comment (nslater) [reply blackbox/docs/configuration/node.txt]: nah, it's fine as is! thx though!
